### DISCUSSION TITLE
random fill stitch displacement

### DISCUSSION THREAD

**Discussion by dirk11182** [inkstitch/inkstitch#2220] [Ideas]
Hello,

is there already a way (I couldn't figure it out) that the stitches of the autofill - method can be randomly displayced? 

I recently compared two embroidery files, one that I made with InkStitch and a commercial one that my wife had bought. Even though the commercial file had quite a few stitches less then my file, it looked a lot prettier (less "holes" and less geometry deformation of the final product). So I searched the files for differences and watched them both in the InkStitch simulator. The only thing that seemed to be different was that the commercial file had some type of randomness while stitching along the same path as my own file. 

so would it be possible to add a feature that can do something similar? When turned on, the only thing that has to happen in addition (to every stitch) is: dont put the actual stitch on its calculated position but make a random displacement (the range could be a parameter in mm for example) around the position or perpendicular to the stitching path. 

I made an image of what I mean because that is much easier to understand but I dont know how to add it to my post, sorry. 

I want to add that InkStitch already is such a wonderful peace of software! Thank you all for your hard work!

**Comment by george-steel** [comment]
I've been meaning to add such a feature like I did with satin stitch, but haven't had time lately.

**Reply by dirk11182** [reply]
That would be great! Is there anything I could do to help?
